app/ydoc-server/src/edits.ts

```diff
@@ -177,11 +182,12 @@ export function stupidFastDiff(oldString: string, newString: string): diff.Diff[
     .concat(commonSuffix ? [[0, commonSuffix]] : [])
 }
 
+/** Return a list of text edits describing how to turn one string into another. */
 export function applyDiffAsTextEdits(
   lineOffset: number,
   oldString: string,
   newString: string,
-): TextEdit[] {
+): readonly TextEdit[] {
```

Comment:
Why `readonly` here? This function returns an array with all objects inside constructed in its body, there is no problem with modifying it by anyone...

Reply:
it's true, i just prefer adding `readonly` unless we need to allow mutation. but it's true that in returns it is totally unnecessary

Reply:
as for parameters, i think it's fine to leave those there - it is a contract that the function does not mutate the array (note that `ReadonlyArray` it is missing methods that mutate arrays, meaning it is a supertype of regular `Array`)